File: side.py
from enum import Enum
"""
The side of the Kalah board a player can choose.
"""

class Side(Enum):
    NORTH
    SOUTH
    """
    @return the side opposite to this one. 
    """
    def opposite(self):
        if self == NORTH:
            return SOUTH
        elif self == SOUTH:
            return NORTH
        else:
            return NORTH

File: javaAgent/src/Side.java
/**
 * The side of the Kalah board a player can choose.
 */
public enum Side
{
    NORTH, SOUTH;

    public static Side mySide;

    /**
     * @return the side opposite to this one.
     */
    public Side opposite()
    {
        switch (this)
        {
            case NORTH: return SOUTH;
            case SOUTH: return NORTH;
            default: return NORTH;  // dummy
        }
    }

    public static boolean myTurn(final Side side) {
        return side == Side.mySide;
    }

    static {
        Side.mySide = Side.SOUTH;
    }
}